crates/bevy_asset/src/lib.rs

```diff
@@ -1,3 +1,106 @@
+//! In the context of game development, an "asset" is a piece of multimedia content which must be loaded from disk and displayed in the game.
+//! Typically, these are authored by artists and designers (in contrast to code),
+//! are relatively large in size, and include everything from textures and models to sounds and music to levels and scripts.
+//!
+//! This presents two main challenges:
+//! - assets take up a lot of memory: simply storing a copy for each instance of an asset in the game would be prohibitively expensive
+//! - loading assets from disk is slow, and can cause the game to stutter or freeze if done in the middle of gameplay
+//!
+//! These problems play into each other: if assets are expensive to store in memory,
+//! larger game worlds will need to load them from disk as needed, ideally without a loading screen!
+//!
+//! Unsurprisingly, the problem of non-blocking asset loading is done using `async`, where background tasks are used to load assets while the game is running.
+//! Bevy coordinates these tasks using the [`AssetServer`], storing each loaded asset in a strongly-typed [`Assets<T>`] collection.
+//! [`Handle`]s serve as an id-based reference to entries in the [`Assets`] collection, allowing them to be cheaply shared between systems,
+//! and providing a way to initialize objects (generally entities) before the required assets are loaded.
+//!
+//! ## Loading and using assets
+//!
+//! The [`AssetServer`] is the main entry point for loading assets.
+//! Typically, you'll use the [`AssetServer::load`] method to load an asset from disk, which returns a [`Handle`].
+//! Note that this method does not attempt to reload the asset if it has already been loaded: as long as the asset hasn't been dropped,
+//! calling [`AssetServer::load`] on the same path will return the same handle.
+//! The handle that's returned will be used to instantiate various [`Component`](bevy_ecs::prelude::Component)s that require asset data to function,
+//! which will then be spawned into the world as part of an entity.
+//!
+//! To avoid assets "popping" into existence, you may want to check that all of the required assets are loaded before transitioning to a new scene.
+//! This can be done by checking the [`LoadState`] of the asset handle using [`AssetServer::is_loaded_with_dependencies`],
+//! which will be [`LoadState::Loaded`] when the asset is ready to use.
+//! Keep track of what you're waiting on using a [`HashSet`] or similar data structure,
+//! poll in your update loop, and transition to the new scene when all assets are loaded.
+//! Bevy's built-in states system can be very helpful for this!
+//!
+//! If we later want to manipulate this asset data (such as for displaying a death animation), we have three options:
+//!
+//! 1. Despawn the entity and spawn a new one with the new asset data.
+//! 2. Change what the handle is pointing to.
+//! 3. Use the [`Assets`] collection to directly modify the asset data.
```

Comment:
```suggestion
//! If we later want to cahnge the asset data a given Component uses (such as changing an entity's material), we have three options:
//! 
//! 1. Change the handle field on the Component to the handle of a different asset
//! 2. Despawn the entity and spawn a new one with the new asset data.
//! 3. Use the [`Assets`] collection to directly modify the current handle's asset data
```

I adjusted some slightly confusing phrasing and put the "most common" option first, to help guide peoples' precedence order. If you accept this change, we'll want to rephrase the following section too.

Reply:
(FYI, there's a typo in the suggestion. `cahnge` → `change`).

Reply:
Fixed. Thanks!